creator-node/src/utils.js

```diff
@@ -244,6 +244,54 @@ async function getAllRegisteredCNodes(libs, logger) {
   return CNodes
 }
 
+/**
+ * Get all Discovery Nodes registered on chain
+ * Fetches from Redis if available, else fetches from chain and updates Redis value
+ * @returns {Promise<Object[]>} array of SP objects with schema { owner, endpoint, spID, type, blockNumber, delegateOwnerWallet }
+ */
+async function getAllRegisteredDNodes({ libs, logger, redis }) {
+  const cacheKey = 'all_registered_dnodes'
+
+  let DNodes
+  try {
+    // Fetch from Redis if present
+    const dnodesList = await redis.get(cacheKey)
+    if (dnodesList) {
+      return JSON.parse(dnodesList)
+    }
+
+    // Else, fetch from chain
+    const discoveryNodes =
+      await libs.ethContracts.ServiceProviderFactoryClient.getServiceProviderList(
+        'discovery-node'
+      )
+
+    // Write fetched value to Redis with 30min expiry
+    await redis.set(
+      cacheKey,
+      JSON.stringify(discoveryNodes),
+      'EX',
+      THIRTY_MINUTES_IN_SECONDS
+    )
+
+    DNodes = discoveryNodes
+  } catch (e) {
+    if (logger) {
```

Comment:
Same note here around logger - you could just destructure it to default to console